### PR TITLE
Baked required deps in docker image for CircleCI.

### PR DESCRIPTION
**Pull request by bwplotka**
This should fix https://github.com/thanos-io/thanos/issues/1483

Also it should massively improve our CI time.

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
Wrongly built image - but can't push fixed one until I land - to weak network.

File: Dockerfile.thanos-ci [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ # Available from https://hub.docker.com/r/circleci/golang/
+FROM circleci/golang:1.12.5
+
+ENV GOBIN=/go/bin
+
+COPY tmp/bin ${GOBIN}
```
Comment (squat) [inline Dockerfile.thanos-ci]: small nit: newline

File: Makefile [hunk 2]
```diff
@@ -46,8 +46,10 @@ ME                ?= $(shell whoami)
 
 # Limited prom version, because testing was not possible. This should fix it: https://github.com/thanos-io/thanos/issues/758
 PROM_VERSIONS           ?= v2.4.3 v2.5.0 v2.8.1 v2.9.2
+PROMS ?= $(GOBIN)/prometheus-v2.4.3 $(GOBIN)/prometheus-v2.5.0 $(GOBIN)/prometheus-v2.8.1 $(GOBIN)/prometheus-v2.9.2
 
 ALERTMANAGER_VERSION    ?= v0.15.2
+ALERTMANAGER 			?= $(GOBIN)/alertmanager-$(ALERTMANAGER_VERSION)
```
Comment (squat) [inline Makefile]: small nit: indentation here is using tabs it seems

File: Makefile [hunk 3]
```diff
@@ -202,15 +207,23 @@ test-local:
...
-	$(call fetch_go_bin_version,github.com/prometheus/alertmanager/cmd/alertmanager,$(ALERTMANAGER_VERSION))
-	$(call fetch_go_bin_version,github.com/minio/minio,$(MINIO_SERVER_VERSION))
+# install-deps installs dependency for e2e tets.
+# It installs supported versions of Prometheus and alertmanager to test against in e2e.
+.PHONY: install-deps
+install-deps: $(ALERTMANAGER) $(MINO_SERVER) $(PROMS)
+	@echo ">>GOBIN=$(GOBIN)"
+
+.PHONY: docker-ci
+docker-ci: install-deps
```
Comment (squat) [inline Makefile]: This comment should go above the target:
> `# To be run by Thanos maintainer.`

File: Makefile [hunk 4]
```diff
@@ -277,6 +290,15 @@ $(GOLANGCILINT):
 $(MISSPELL):
 	$(call fetch_go_bin_version,github.com/client9/misspell/cmd/misspell,$(MISSPELL_VERSION))
 
+$(ALERTMANAGER):
+	$(call fetch_go_bin_version,github.com/prometheus/alertmanager/cmd/alertmanager,$(ALERTMANAGER_VERSION))
+
+$(MINO_SERVER):
```
Comment (squat) [inline Makefile]: This variable is never defined, which means that the target isn't executed and minio isn't installed :/

File: Dockerfile.thanos-ci [hunk 5]
```diff
@@ -0,0 +1,6 @@
+ # Available from https://hub.docker.com/r/circleci/golang/
```
Comment (adrien-f) [inline Dockerfile.thanos-ci]: Extra space before the `#` ?

File: Makefile [hunk 6]
```diff
@@ -202,15 +207,23 @@ test-local:
...
-# test-deps installs dependency for e2e tets.
-# It installs current Thanos, supported versions of Prometheus and alertmanager to test against in e2e.
-.PHONY: test-deps
-test-deps:
-	@echo ">> install thanos GOOPTS=${GOOPTS}"
-	@go install github.com/thanos-io/thanos/cmd/thanos
-	$(foreach ver,$(PROM_VERSIONS),$(call fetch_go_bin_version,github.com/prometheus/prometheus/cmd/prometheus,$(ver)))
-	$(call fetch_go_bin_version,github.com/prometheus/alertmanager/cmd/alertmanager,$(ALERTMANAGER_VERSION))
-	$(call fetch_go_bin_version,github.com/minio/minio,$(MINIO_SERVER_VERSION))
+# install-deps installs dependency for e2e tets.
```
Comment (adrien-f) [inline Makefile]:
```suggestion
# install-deps installs dependency for e2e tests.
```

File: Makefile (hunk 4 above)
Comment (bwplotka) [reply Makefile]: :man_facepalming:

Comment (bwplotka) [reply Makefile]: fixed

Comment (squat) [reply Makefile]: here the variable is called `MINO_SERVER` and up top it's defined as `MINIO_SERVER` so we have the same issue. We just need to fix the misspelling here for it to work

**Comment by domgreen** [comment]
CI currently failing with:
```
    store_gateway_test.go:51: spinup failed: start minio: start: exec: "minio-RELEASE.2018-10-06T00-15-16Z": executable file not found in $PATH
```

**Comment by squat** [comment]
@domgreen yes, the issue is this: https://github.com/thanos-io/thanos/pull/1484#discussion_r319928970

File: Makefile (hunk 4 above)
Comment (bwplotka) [reply Makefile]: fixed v2.0

Comment (squat) [reply Makefile]: :sunglasses:

**Comment by squat** [comment]
:tada:

**Pull request merged**